code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -330,6 +330,32 @@
 /mob/living/silicon/ai/cancel_camera()
 	view_core()
 
+GLOBAL_LIST_EMPTY(empty_playable_ai_cores)
+
+/proc/roundstart_spawn_empty_ai_if_needed()
+	for(var/obj/effect/landmark/start/S in GLOB.landmarks_list)
+		if(S.name != "AI")
+			continue
+		if(locate(/mob/living) in S.loc)
+			continue
+		GLOB.empty_playable_ai_cores += new /obj/structure/AIcore/deactivated(get_turf(S))
+
+	return 1
+
+// Before calling this, make sure an empty core exists, or this will no-op
+/mob/living/silicon/ai/proc/moveToEmptyCore()
+	if(!GLOB.empty_playable_ai_cores.len)

```

Comment:
This should probably throw an alert at admins or an exception or something. Will be a bitch to debug if we forget somewhere down the line.